bindings/python/nrm/setup.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2019 UChicago Argonne, LLC.
+# (c.f. AUTHORS, LICENSE)
+#
+# This file is part of the libnrm project.
+# For more info, see https://github.com/anlsys/libnrm
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import signal
+import subprocess
+
+
+class NRMBinary:
+    def __init__(self, name, mustkill=False, waitready=None):
+        self.name = name
+        self.mustkill = mustkill
+        self.waitready = waitready
+
+    def launch(self, options, args):
+        self.abspath = (options["prefix"] / self.name).absolute()
+        self.stdout = open(
+            options["output"] / (self.name + "-stdout.log"), "w+"
+        )
+        self.stderr = open(
+            options["output"] / (self.name + "-stderr.log"), "w+"
+        )
+        assert self.stdout is not None
+        assert self.stderr is not None
+        popencmd = []
+        # support for LOG_COMPILER and LOG_FLAGS
+        popencmd += os.environ.get("LOG_COMPILER", "").split()
+        popencmd += os.environ.get("LOG_FLAGS", "").split()
+        popencmd += [self.abspath]
+        # add the rest of the command line arguments
+        popencmd += args
+        self.proc = subprocess.Popen(
+            popencmd, stdin=None, stdout=self.stdout, stderr=self.stderr
+        )
+        assert self.proc
+
+        # in most cases, we should wait before launching something else
+        if self.waitready:
+            self.waitready(options)
+
+    def isdead(self):
+        return self.proc.poll() is not None
+
+    def kill(self):
+        if self.mustkill:
+            os.kill(self.proc.pid, signal.SIGKILL)
+        else:
+            os.kill(self.proc.pid, signal.SIGTERM)
+        self.proc.wait()
+        self.stdout.close()
+        self.stderr.close()
+
+
+def nrmd_waitready(options):
+    nrmc_cmd = [(options["prefix"] / "nrmc").absolute(), "-q", "connect"]
+    subprocess.run(nrmc_cmd, check=True)
+
+def dummy_timebuffer(options):
+    import time
+    time.sleep(1)
```

Comment:
Not a good wait function. Looks at the nrm-setup binary, we need retries, an exception on failure, and to wait for the dummy sensor to appear in list_sensors.